src/tests/unit/tests/background/stores/global/permissions-state-store.test.ts

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import { PermissionsStateActions } from 'background/actions/permissions-state-actions';
+import { PermissionsStateStore } from 'background/stores/global/permissions-state-store';
+import { StoreNames } from 'common/stores/store-names';
+import { PermissionsStateStoreData } from 'common/types/store-data/permissions-state-store-data';
+import { StoreTester } from 'tests/unit/common/store-tester';
+
+describe('PermissionsStateStoreTest', () => {
+    const expectedDefaultStoreData: PermissionsStateStoreData = {
+        hasAllPermissions: false,
+    };
+
+    test('constructor', () => {
+        const testSubject = new PermissionsStateStore(new PermissionsStateActions());
+
+        expect(testSubject).toBeDefined();
+    });
+
+    test('getId', () => {
+        const testSubject = new PermissionsStateStore(new PermissionsStateActions());
+
+        expect(testSubject.getId()).toEqual(StoreNames[StoreNames.PermissionsStateStore]);
+    });
+
+    test('initialize sets store state to default state', () => {
+        const testSubject = new PermissionsStateStore(new PermissionsStateActions());
+        testSubject.initialize();
+
+        expect(testSubject.getState()).toEqual(expectedDefaultStoreData);
+    });
+
+    test('getDefaultState returns expected default store state', () => {
+        const testSubject = new PermissionsStateStore(new PermissionsStateActions());
+        testSubject.initialize();
+
+        expect(testSubject.getDefaultState()).toEqual(expectedDefaultStoreData);
+    });
+
+    test('on getCurrentState', () => {
+        const initialState = createPermissionsState();
+        const finalState = createPermissionsState();
+
+        createStoreTesterForPermissionsStateActions('getCurrentState').testListenerToBeCalledOnce(initialState, finalState);
+    });
+
+    test('on setPermissionsState updates state when it has changed', () => {
+        const initialState = expectedDefaultStoreData;
+        const finalState = { hasAllPermissions: true };
+
+        createStoreTesterForPermissionsStateActions('setPermissionsState')
+            .withActionParam(true)
```

Comment:
should the param be extracted to a `const` and used here to call `.withActionParam` and also to define the `finalState`?
This way, it's more clear that the `true` value is closely related and not just a coincidence.